File: mooc-2024/part-4/4-define-iteration/11-the-shortest-in-the-list/main.py
def shortest(l):
    len_list = list(map(len, l))
    min_len = min(len_list)
    return l[len_list.index(min_len)]

my_list = ["first", "second", "fourth", "eleventh"]

my_list = ["first", "second", "fourth", "eleventh"]

result = shortest(my_list)
print(result)
my_list = ["adele", "mark", "dorothy", "tim", "hedy", "richard"]

result = shortest(my_list)
print(result)

File: mooc-2024/part-4/4-define-iteration/11-the-shortest-in-the-list/main.c
#include <stdio.h>
#include <string.h>
#include "../../../../lib/input.h"

int main() {
    int len = 4;
    int min_len = 999;
    char *min_str = NULL;
    char arr[][50] = {"the", "second", "fourth", "eleventh"};
    
    for (int idx = 0; idx < len; idx++) {
        if (strlen(arr[idx]) < min_len) {
            min_len = strlen(arr[idx]);
            min_str = arr[idx];
        }
    }

    printf("The minimum length: %d\n", min_len);
    printf("%s\n", min_str);

    return 0;
}